### PR TITLE
Allow subdirectory server configuration, e.g. https://example.com/isso

### PR DESCRIPTION
**Pull request by craigphicks**
# Allow sudirectory server configuration, , e.g. https://example.com/isso

## Description

When the isso root location was 

    https://example.com/isso

the administration pages did not work correctly.  The changes in this pull request fix that and allow proper working with or without a subdirectory.  E.g. both of these become valid configurations:

    https://example.com
    https://example.com/isso

The subdirectory case only requires the subdirectory to be passed as a parameter in the proxy call.  Here is an example *nginx* setup:

    location /isso {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Script-Name /isso;      // <---- critical for subdirectory config
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_pass http://localhost:1234;
    }
 
If the line `proxy_set_header X-Script-Name /isso;` is omitted, communication with *isso* will fail.

## Testing 

Given the blog address "https:\example.com", *isso* was tested using the following sites:

    https://example.com/isso
    https://another.com
    https://another.com/isso

## Other bugs found, not fixed yet

The *active* and *delete* URLs sent by the *isso* mail notification system were found to fail under the following conditions, with the follow symptoms:

1. *Condition:* Blog and Isso servers differ.  E.g.  `https://example.com/` and `https://another.com/`.  
    *Symptom:* The *active/delete* URLs use the blog server instead of the isso server.
2. *Condition:* Isso is cofigured as a subdirectory, e.g. `https://example.com/isso`.  
    *Symptom:* The `isso` part is not included in the URL.
3. *Condition:* Both 1 and 2 together.
    *Symptom:* Both

I will try to address these mail bugs later.

### CODE REVIEW

**Comment by craigphicks** [comment]
Now I have read 
https://github.com/posativ/isso/issues/420  
which proposes a configuration setting for the isso server public url.  
That would seemingly remove the need for the changes in this pull.

File: isso/views/comments.py [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ def login(self, env, req):
             response = redirect(re.sub(
                 r'/login$',
                 '/admin',
-                get_current_url(env, strip_querystring=True)
+                get_current_url(env, strip_querystring=True) #iwozere
```
Comment (blatinier) [inline isso/views/comments.py]: This comment should go away.

File: isso/templates/admin.html [hunk 2]
```diff
@@ -32,16 +32,16 @@
...
+        url: "" + isso_host_script + "/id/" + com_id + "/" + action + "/" + hash,
         success: function(){
             fade(document.getElementById("isso-" + com_id));
         }});
 }
-function edit(com_id, hash, author, email, website, comment) {
+function edit(com_id, hash, author, email, website, comment, isso_host_script) {
   ajax({method: "POST",
-        url: "/id/" + com_id + "/edit/" + hash,
+        url: "" + isso_host_script + "/id/" + com_id + "/edit/" + hash,
```
Comment (blatinier) [inline isso/templates/admin.html]: The first `"" + ` is useless

File: isso/templates/admin.html [hunk 3]
```diff
@@ -32,16 +32,16 @@
         op -= op * 0.1;
     }, 10);
 }
-function moderate(com_id, hash, action) {
+function moderate(com_id, hash, action, isso_host_script) {
   ajax({method: "POST",
-        url: "/id/" + com_id + "/" + action + "/" + hash,
+        url: "" + isso_host_script + "/id/" + com_id + "/" + action + "/" + hash,
```
Comment (blatinier) [inline isso/templates/admin.html]: The first `"" + ` is useless

**Comment by craigphicks** [comment]
@blatinier - I have updated and pushed.  [This page of differences](https://github.com/posativ/isso/pull/458/files) now looks correct.

**Comment by craigphicks** [comment]
Something must be added to documentation about server setting for the subdirectory case:  E.g. the below information.  That can be added in another pull if this one is ready to go.  Or it can be added in this pull.

----------------------

The subdirectory case only requires the subdirectory to be passed as a parameter in the proxy call. Here is an example nginx setup:

location /isso {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Script-Name /isso;      // <---- critical for subdirectory config
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_pass http://localhost:1234;
}

If the line proxy_set_header X-Script-Name /isso; is omitted, communication with isso will fail.

**Comment by craigphicks** [comment]
I see that documentation about `proxy_set_header X-Script-Name /isso;` already exists [under docs](https://github.com/posativ/isso/blob/master/docs/docs/setup/sub-uri.rst).

**Comment by craigphicks** [comment]
I have modified `isso/views/comments.py` to follow the policy started in [pull 443](https://github.com/posativ/isso/pull/443), which that the so-called 'public_endpoint' will be read from the conf file *general/public_endpoint*, and if that is not present it will be taken from the internal variable `local.host`.

This pull 458 originally introduced  `local.host` usage to to enable isso service as a subdirectory of the host URL, e.g. `https://example.com/isso`, where `/isso` is the subdirectory.  In the particular test case *isso* is proxied from an *nginx* server, with the configuration as shown above.

The purpose of the config *general/public_endpoint* setting option is as fallback in case, due to an unusual setup, or whatever.

It is worth adding that there already exists a config setting *general/name* which may have duplicate functionality with *general/public_endpoint*.  When setting up a second isso, and serving from an *nginx* location 'isso-2': 

```
location /isso-2 {
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Script-Name /isso-2;      // <---- critical for subdirectory config
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_pass http://localhost:1234;
}

```
I found it was necessary to set 

```
[general]
name = isso-2
```

in order to get the connection to work.  Indeed the *isso* documentation on [Multiple Sites](https://posativ.org/isso/docs/setup/multiple-sites/) explains that it is necessary to specify the subdirectory in config under *general/name*.

**Pull request merged**